src/main/java/com/hivemq/extensions/iteration/Chunker.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright 2019-present HiveMQ GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hivemq.extensions.iteration;
+
+import com.google.common.base.Preconditions;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.util.concurrent.Futures;
+import com.google.common.util.concurrent.ListenableFuture;
+import com.google.common.util.concurrent.MoreExecutors;
+import com.hivemq.configuration.service.InternalConfigurations;
+import com.hivemq.extension.sdk.api.annotations.NotNull;
+
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import java.util.Map;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+/**
+ * @author Georg Held
+ */
+@Singleton
+public class Chunker {
+
+    private final int bucketCount;
+
+    @Inject
```

Comment:
what do you inject here?

Reply:
From the Guice coding guidelines:

> By default Guice does not require you to annotate your public constructor
with @Inject - if there's only one such public constructor then it will
automatically pick that one. Even when there are multiple constructors then
you can use a constructor binding to tell Guice which one to use and avoid
the ambiguity.
>
>Using @Inject is still recommended for code under your control because it
documents that a class was written with injection in mind.